eo-maven-plugin/src/test/java/org/eolang/maven/testapi/FakeMaven.java

```diff
@@ -0,0 +1,195 @@
+ * SOFTWARE.
+ */
+package org.eolang.maven.testapi;
+
+import com.yegor256.tojos.Tojo;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.maven.plugin.AbstractMojo;
+import org.eolang.maven.AssembleMojo;
+import org.eolang.maven.Catalogs;
+import org.eolang.maven.Home;
+import org.eolang.maven.Moja;
+
+/**
+ * Fake maven workspace that executes Mojos in order to test
+ * their behaviour and results.
+ * @since 0.28.12
+ * @todo #1417:90min Make `FakeMaven.exec()` return a `HashMap` with all
+ *  files created in the directory and their relative names. Then, we can assert on this hash map.
+ */
+public final class FakeMaven {
+
+    /**
+     * Default eo program id.
+     */
+    private static final String PROGRAM_ID = "foo.x.main";
+
+    /**
+     * Default eo program path.
+     */
+    private static final String PROGRAM_PATH = "foo/x/main.eo";
+
+    /**
+     * Default eo-foreign.csv file format.
+     */
+    private static final String FOREIGN_FORMAT = "csv";
+
+    /**
+     * Default eo-foreign.csv file path.
+     */
+    private static final Path FOREIGN_PATH = Paths.get("eo-foreign.csv");
+
+    /**
+     * Test workspace where we place all programs, files, compilation results, etc.
+     */
+    private final Home workspace;
+
+    /**
+     * Mojos params.
+     */
+    private final Map<String, Object> params;
+
+    /**
+     * Attributes for eo.foreign.*.
+     */
+    private final Map<String, Object> attributes;
+
+    /**
+     * Path to a program in workspace.
+     */
+    private Path prog;
+
+    /**
+     * The main constructor.
+     *
+     * @param workspace Test temporary directory.
+     */
+    public FakeMaven(final Path workspace) {
+        this.workspace = new Home(workspace);
+        this.params = this.defaultParams();
+        this.attributes = new HashMap<>();
+    }
+
+    /**
+     * Adds eo program to a workspace.
+     * @param program Program as a raw string.
+     * @return Workspace with eo program.
+     * @throws IOException If can't save eo program in workspace.
+     */
+    public FakeMaven forProgram(final String... program) throws IOException {
+        final Path path = Paths.get(FakeMaven.PROGRAM_PATH);
+        this.workspace.save(String.join("\n", program), path);
+        this.prog = path;
+        return this;
+    }
+
+    /**
+     * Sets parameter for execution.
+     *
+     * @param key Parameter name
+     * @param value Parameter value
+     * @return The same maven instance.
+     */
+    public FakeMaven with(final String key, final Object value) {
+        this.params.put(key, value);
+        return this;
+    }
+
+    /**
+     * Sets tojo attribute.
+     *
+     * @param attribute Tojo attribute.
+     * @param value Attribute value.
+     * @return The same maven instance.
+     */
+    public FakeMaven with(final TojoAttribute attribute, final Object value) {
```

Comment:
@volodya-lombrozo can't you just rename this method to `withTojo` (or something similar) and get rid of this supplementary class?

Reply:
@yegor256 fixed. `with` -> `withTojoAttribute`